lib/http2/frame.c

```diff
@@ -53,6 +53,7 @@ int h2o_http2_update_peer_settings(h2o_http2_settings_t *settings, const uint8_t
             SET(MAX_CONCURRENT_STREAMS, max_concurrent_streams, 0, UINT32_MAX, 0);
             SET(INITIAL_WINDOW_SIZE, initial_window_size, 0, 0x7fffffff, H2O_HTTP2_ERROR_FLOW_CONTROL);
             SET(MAX_FRAME_SIZE, max_frame_size, 16384, 16777215, H2O_HTTP2_ERROR_PROTOCOL);
+            SET(DEPRECATE_HTTP2_PRIORITIES, deprecate_http2_priorities, 0, 1, H2O_HTTP2_ERROR_PROTOCOL);
```

Comment:
I am not sure if we need to send this settings parameter now; aren't we only recognizing the signal from the origin (sent as a header field)? I think we should not be sending this settings parameter to the client until we support accepting the header field (or whatever the priority draft would define) from the client.

Reply:
Okay, it makes sense. I'll drop this patch for now.

Reply:
Turns out we both misunderstood this code snippet.

This is actually about *accepting* `SETTINGS_DEPRECATE_HTTP2_PRIORITIES` from a client. At this moment this doesn't do any meaningful work other than checking the value range. See the patch `[http2] Accept SETTINGS_DEPRECATE_HTTP2_PRIORITIES` for details.

FYI, the patch to *send* `SETTINGS_DEPRECATE_HTTP2_PRIORITIES` has already been dropped from this PR, so it is not sent to a client. If you prefer, I can also drop this patch (`[http2] Accept SETTINGS_DEPRECATE_HTTP2_PRIORITIES`).

Reply:
Ah. Thank you for the explanation.

I think it might be a good idea to drop all code related to SETTINGS_DEPRECATE_HTTP2_PRIORITIES, even though the code looks fine and I appreciate your efforts.

The reasons are:
* we do not use the flag in any ways
* the specification might change
* we might use the value differently. For example, rather than retaining a boolean variable, we might use the existence (or lack of) the settings parameter to initiate different scheduler.

We can resurrect the code when necessary.

Reply:
👍 Totally makes sense. I'll drop this patch. Thanks!

Reply:
Done.